### PR TITLE
chore: enable easier electron builds by hosting some packages

### PR DESCRIPTION
**Pull request by a-b-r-o-w-n**
## Description

- Introduces `@botframework-composer` npm scope for some shared packages that can be installed inside of plugins. This reduces the yarn workspace magic and fixes electron builds due to missing dependencies.
- Removes `@bfc/shared` dependency from `@bfc/extension-client` to ease installation outside of the workspace and in preparation for publishing.

## Task Item

#minor

### CODE REVIEW

File: Composer/packages/client/src/components/PluginHost/PluginHost.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@
 /** @jsx jsx */
 import { jsx, SerializedStyles } from '@emotion/core';
 import React, { useState, useEffect, useRef } from 'react';
-import { Shell } from '@bfc/types';
+import { Shell } from '@botframework-composer/types';
```
Comment (tonyanziano) [inline Composer/packages/client/src/components/PluginHost/PluginHost.tsx:7]: maybe I missed it, but I don't see `@botframework-composer/types` in `dependencies` or `devDependencies` of client

File: Composer/plugins/azurePublish/yarn.lock [hunk 2]
```diff
@@ -178,11 +179,19 @@
...
   dependencies:
+    "@botframework-composer/types" "*"
     format-message "6.2.3"
     json-schema "^0.2.5"
     nanoid "^3.1.3"
     nanoid-dictionary "^3.0.0"
 
+"@botframework-composer/types@*", "@botframework-composer/types@0.0.1":
+  version "0.0.1"
+  resolved "https://botbuilder.myget.org/F/botbuilder-v4-js-daily/npm/@botframework-composer/types/-/@botframework-composer/types-0.0.1.tgz#117b86bf8e3bf43567471d436eb1c2f0ffcd6bb8"
```
Comment (tonyanziano) [inline Composer/plugins/azurePublish/yarn.lock:190]: is this expected?

**Review by tonyanziano — CHANGES_REQUESTED** [review]

File: Composer/packages/client/src/components/PluginHost/PluginHost.tsx (hunk 1 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/components/PluginHost/PluginHost.tsx:7]: Good catch.

File: Composer/plugins/azurePublish/yarn.lock (hunk 2 above)
Comment (a-b-r-o-w-n) [reply Composer/plugins/azurePublish/yarn.lock:190]: Which part? The registry?

Comment (tonyanziano) [reply Composer/plugins/azurePublish/yarn.lock:190]: yeah the registry

Comment (a-b-r-o-w-n) [reply Composer/plugins/azurePublish/yarn.lock:190]: There is an `.npmrc` in this directory that sets that registry. so yes?

Comment (tonyanziano) [reply Composer/plugins/azurePublish/yarn.lock:190]: didn't see that

**Review by tonyanziano — APPROVED** [review]

File: Composer/packages/adaptive-form/src/components/fields/RegexIntentField.tsx [hunk 3]
```diff
@@ -26,14 +28,15 @@ function getRegexIntentPattern(currentDialog: DialogInfo, intent: string): strin
...
   const { currentDialog, shellApi } = useShellApi();
-  const [localValue, setLocalValue] = useState(getRegexIntentPattern(currentDialog, intentName));
+  const formData = useFormData();
+  const [localValue, setLocalValue] = useState(getRegexIntentPattern(formData, intentName));
 
   // if the intent name changes or intent names in the regex patterns
   // we need to reset the local value
   useEffect(() => {
-    const pattern = getRegexIntentPattern(currentDialog, intentName);
+    const pattern = getRegexIntentPattern(formData, intentName);
```
Comment (yeze322) [inline Composer/packages/adaptive-form/src/components/fields/RegexIntentField.tsx:37]: refs #4653

**Pull request merged**